src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -67,11 +67,23 @@ internal class SourceLocalSymbol : LocalSymbol
             _locations = ImmutableArray.Create<Location>(identifierToken.GetLocation());
         }
 
-        internal Binder Binder
+        /// <summary>
+        /// Binder that owns the scope for the local, the one that returns it in its <see cref="Binder.Locals"/> array.
+        /// </summary>
+        internal Binder ScopeBinder
         {
-            get { return binder; }
+            get { return _scopeBinder; }
         }
 
+        /// <summary>
+        /// Binder that should be used to bind type syntax for the local.
+        /// </summary>
+        internal Binder TypeSyntaxBinder
```

Comment:
To clarify, addition TypeSyntaxBinder property wasn't necessary, addition of initializer binder for LacalWithInitializer was necessary for correct behavior.